sonic_platform_base/sonic_xcvr/api/public/c_cmis.py

```diff
@@ -150,6 +152,46 @@ def set_tx_power(self, tx_power):
         time.sleep(1)
         return status
 
+    def util_status_check_with_retry(self, retry, sleep_time, reg_field, expected_status):
+        '''
+         Utility function which retries for expected status with timeout and sleep between retries.
+         Input args: retry count, sleep time between retries, register to read for status, expected status to compare.
+         output : 0 - register read mactches expected status, 1 - retry timed out.
+         '''
+        time.sleep(sleep_time)
+        status = 0
+        while(retry > 0):
+            status = self.xcvr_eeprom.read(reg_field)
+            if (status == expected_status):
+                return 0
+            retry = retry-1
+            time.sleep(sleep_time)
+        if (retry <= 0) and (status != expected_status):
+            return 1
+
+    def freeze_vdm_stats(self):
+        '''
+        This function freeze all the vdm statistics reporting registers.
+        When raised by the host, causes the module to freeze and hold all 
+        reported statistics reporting registers (minimum, maximum and 
+        average values)in Pages 24h-27h.
+        '''
+        self.xcvr_eeprom.write(consts.VDM_CONTROL, VDM_FREEZE)
+        #Retry for freeze done status
+        return self.util_status_check_with_retry(VDM_FREEZE_UNFREEZE_DONE_RETRY, SLEEP_10MS, consts.VDM_FREEZE_DONE, 1)
+
+    def unfreeze_vdm_stats(self):
```

Comment:
@jaganbal-a After writing to un-Freeze, we should wait at max tVDMF Timeout = 10msec and if still VDM_UNFREEZE_DONE is not SET return failure